components/hab/src/main.rs

```diff
@@ -841,6 +843,51 @@ fn sub_pkg_sign(ui: &mut UI, m: &ArgMatches<'_>) -> Result<()> {
     command::pkg::sign::start(ui, &pair, &src, &dst)
 }
 
+fn sub_pkg_bulkupload(ui: &mut UI, m: &ArgMatches<'_>) -> Result<()> {
+    let upload_dir = bulkupload_dir_from_matches(m);
+    // upload_dir is required via clap and exists
+    let artifact_path = upload_dir.as_ref().unwrap().join("artifacts");
+    let key_path = upload_dir.as_ref().unwrap().join("keys");
+    let url = bldr_url_from_matches(&m)?;
+    let additional_release_channel = channel_from_matches(&m);
+    let force_upload = m.is_present("FORCE");
+    let auto_build = if m.is_present("AUTO_BUILD") {
+        BuildOnUpload::PackageDefault
+    } else {
+        BuildOnUpload::Disable
+    };
+    let token = auth_token_param_or_env(&m)?;
+    let options = glob::MatchOptions { case_sensitive:              true,
+                                       require_literal_separator:   true,
+                                       require_literal_leading_dot: true, };
+    let artifact_paths =
+        vec_from_glob_with(&artifact_path.join("*.hart").display().to_string(), options);
```

Comment:
OPTIONAL: the type returned is inherent in the signature, so I think the name could be simplified to `glob_with`.

Reply:
Since `glob::glob_with` is already a function I'm importing I went with the `vec_from_glob_with`.

Reply:
Yeah, not a big deal either way. My recommendation would be to remove the `use glob::glob_with;`, since you can still refer to `glob::glob_with` using that fully-qualified name. I think it makes things clearer to call into other crates with an at least partially qualified name and leave unqualified function names for only crate local things. Having a local `glob_with` that calls `glob::glob_with` internally is perfectly acceptable in my opinion.

But as I said, not a big deal, so do what you like.